watertap/examples/chemistry/chem_scaling_utils.py

```diff
@@ -0,0 +1,119 @@
+# "https://github.com/watertap-org/watertap/"
+#
+###############################################################################
+
+"""
+    This is a set of scaling utility helper functions used by the chemistry tests.
+"""
+from idaes.core.util import scaling as iscale
+from pyomo.environ import value
+
+__author__ = "Austin Ladshaw"
+
+## Helper function for setting eps values associated with solubility_product functions
+# NOTE: Function does nothing if no solubility reactions are present
+def _set_eps_vals(rxn_params, rxn_config, factor=1e-2):
+    # For solubility reactions, have to set the eps value
+    if (hasattr(rxn_params,"equilibrium_reaction_idx")):
+        for rid in rxn_params.equilibrium_reaction_idx:
+            # Grab the 'k_eq_ref' value from the reaction config
+            scale = rxn_config["equilibrium_reactions"][rid]["parameter_data"]["k_eq_ref"][0]
+
+            # NOTE: ONLY certain functions have an eps value that we need to set
+            if (hasattr(rxn_params.component("reaction_"+rid),"eps")):
+                # highest allowable value for setting eps based on k_eq_ref
+                if scale < 1e-16:
+                    rxn_params.component("reaction_"+rid).eps.value = scale*factor
+                else:
+                    rxn_params.component("reaction_"+rid).eps.value = 1e-16*factor
+
+## Helper function for setting scaling factors for equilibrium reactions
+def _set_equ_rxn_scaling(unit, rxn_config, min_scale=1e-3):
+    #Add scaling factors for reactions (changes depending on if it is a log form or not)
+    for i in unit.control_volume.equilibrium_reaction_extent_index:
+        # i[0] = time, i[1] = reaction
+
+        # Grab the 'k_eq_ref' value from the reaction config
+        scale = max(min_scale, rxn_config["equilibrium_reactions"][i[1]]["parameter_data"]["k_eq_ref"][0])
+        iscale.set_scaling_factor(unit.control_volume.equilibrium_reaction_extent[0.0,i[1]], 10/scale)
+        iscale.constraint_scaling_transform(
+            unit.control_volume.reactions[0.0].equilibrium_constraint[i[1]], 0.1)
+
+## Helper function for setting scaling factors for inherent reactions
+def _set_inherent_rxn_scaling(unit, thermo_config, min_scale=1e-3):
+    #Add scaling factors for reactions (changes depending on if it is a log form or not)
+    for i in unit.control_volume.inherent_reaction_extent_index:
+        # i[0] = time, i[1] = reaction
+
+        # Grab the 'k_eq_ref' value from the thermo config
+        scale = max(min_scale, thermo_config["inherent_reactions"][i[1]]["parameter_data"]["k_eq_ref"][0])
+        iscale.set_scaling_factor(unit.control_volume.inherent_reaction_extent[0.0,i[1]], 10/scale)
+        iscale.constraint_scaling_transform(unit.control_volume.properties_out[0.0].
+                inherent_equilibrium_constraint[i[1]], 0.1)
+
+## Helper function for setting scaling factors for rate reactions
+def _set_rate_rxn_scaling(rxn_params, unit, min_scale=1e-3):
```

Comment:
`min_scale` isn't used. Should it be?
```suggestion
def _set_rate_rxn_scaling(rxn_params, unit):
```

Reply:
I haven't yet fully worked out how I actually want to apply scaling for this type of reaction. What I have right now works, but I have done only very limited testing with it. I anticipate it needing some 'min_scale' arg later, so I put one in the function call. I may add a TODO statement here as a reminder.